### PR TITLE
Fleshing out KNN Classifier Documentation

### PR DESCRIPTION
**Pull request by rmccall84**
Fixes https://github.com/numenta/nupic/issues/2232

This is the first step in addressing several issues with the KNN Classifier. The documentation represents my current best understanding of the module. Here I focus on docstring and formatting changes, not code changes. Even after these changes, some aspects of the code may still be insufficiently documented. Adding documentation to a construct does not imply an endorsement, rather the goal here is to better understand the intent of the code. Future PRs will address coding issues, API design, unittesting, etc.

### CODE REVIEW

File: nupic/algorithms/KNNClassifier.py [hunk 1]
```diff
@@ -297,27 +298,34 @@ def _removeRows(self, rowsToRemove):
...
+                                  (both ON and OFF bits present) array.
+                                  Otherwise, if isSparse > 0, this should be a
+                                  list of non-zero indices.
+    @param inputCategory: (int)   The category index of the training pattern.
+    @param partitionId:   (int)   UNKNOWN
+    @param isSparse:      (int)   If 0, the input pattern is a dense
+                                  representation. If > 0, the input pattern is a
+                                  list of non-zero indices and isSparse is the
+                                  length of the dense representation.
+    @param rowID:         (int)   UNKNOWN
```
Comment (breznak) [inline nupic/algorithms/KNNClassifier.py]: rather mark these TODO? or even file an issue for it?

Comment (rmccall84) [inline nupic/algorithms/KNNClassifier.py]: thx. I'll add them to this issue: https://github.com/numenta/nupic/issues/2198

**Comment by rmccall84** [comment]
@scottpurdy @subutai please CR

**Comment by scottpurdy** [comment]
Since this is mostly about formatting, I'll point out a couple things from our style guide (which references PEP 8 and PEP 257)

- Take a look at the recommended one-line and multi-line doc string formats: https://www.python.org/dev/peps/pep-0257/
- Don't use white space to align separate statements. I think in the same spirit that you should avoid aligning documentation for different parameters in the doc strings. This means you have to do a bunch of unecessary formatting when you change one of the parameters. https://www.python.org/dev/peps/pep-0008/#pet-peeves

**Comment by rmccall84** [comment]
@scottpurdy integration tests still pass. please merge if you accept

**Pull request merged**